src/core/namespace.c

```diff
@@ -810,6 +845,12 @@ static void drop_duplicates(MountList *ml) {
                         continue;
                 }
 
+                if (f->mode == MOUNT_EXTENSION_IMAGE && verity_has_duplicates(ml, f)) {
```

Comment:
if you add the if check as i suggest above then you can drop the part left of the && here

Reply:
added an assert, the check here is clearer I think